nixos/modules/services/networking/ntp/ntpd.nix

```
@@ -132,6 +132,7 @@ in
 
         preStart =
           ''
+            # shellcheck disable=SC2174
             mkdir -m 0755 -p ${stateDir}
```

```suggestion
            mkdir -m 0755 ${stateDir} || true
```
what do you think about this?

---

Perhaps this code can be removed completely by utilizing `StateDirectory` for the service. `stateDir` is hardcoded to be `/var/lib/ntp`

---

Yeah, completely missed that!

---

Good call.